File: 1748-sum-of-unique-elements/1748-sum-of-unique-elements.cpp
class Solution {
public:
    int sumOfUnique(vector<int>& nums) {
        size_t n = nums.size();

        int freq[101] = {0};
        int sum = 0;

        for(const int num : nums){
            if(freq[num] == 0){
                sum += num;
                freq[num]++;
            }else{
                if(freq[num] == 1) sum -= num;
                freq[num]++;
            }
        }

        return sum;
    }
};

File: 1748-sum-of-unique-elements/1748-sum-of-unique-elements.go
func sumOfUnique(nums []int) int {
    var n int8 = int8(len(nums))

    var freq [101]int8

    var sum int = 0

    for i := range int(n) {
        if freq[nums[i]] == 0 {
            sum += nums[i]
            freq[nums[i]]++
        }else{
            if freq[nums[i]] == 1 {
                sum -= nums[i]
            }
            freq[nums[i]]++
        }
    }

    return sum
}